src/Compilers/Test/Utilities/VisualBasic/ParserTestUtilities.vb

```diff
@@ -75,12 +77,16 @@ Friend Module ParserTestUtilities
 
         ' Verify Errors
         If expectedDiagnostics Is Nothing Then
-            Dim errors As New StringBuilder()
-            AppendSyntaxErrors(tree.GetDiagnostics(), errors)
-            Assert.False(root.ContainsDiagnostics, errors.ToString())
-            Assert.Equal(root.ContainsDiagnostics, errors.Length > 0)
+            Dim errors = PooledStringBuilderPool.Allocate()
+            AppendSyntaxErrors(tree.GetDiagnostics(), errors.Builder)
+            Assert.Equal(root.ContainsDiagnostics, errors.Builder.Length > 0)
+            If root.ContainsDiagnostics Then
```

Comment:
>If root.ContainsDiagnostics Then [](start = 12, length = 32)

It doesn't look like this ```If``` statement is equivalent to the ```Assert.False``` that it is replacing. The assert would fail if ```root.ContainsDiagnostics``` is true, but the ```If``` simply frees the builder in this case.